### DISCUSSION TITLE
Automatically scroll to the next entry

### DISCUSSION THREAD

**Discussion by BartNijland91** [patrykandpatrick/vico#352] [Ideas]
Hi!

Is there any way to automatically scroll to the next entry? Currently I have some dirty hack that gets the width between first and second entry, but that's not really the way to go. Is this possible and/or are there any plans to build this?

**Comment by patrickmichalik** [comment]
Hello! There’s no such feature at the moment, but we’ll look into adding it. Thanks for the suggestion!

**Comment by patrickmichalik** [comment]
[Vico 2.0.0 Alpha 9](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.9) adds such functionality. To scroll to the position of a specific entry, use `Scroll.Absolute.x`, passing in the entry’s _x_ coordinate, which you can align via the `bias` parameter. Depending on the specifics of the desired result, it can be helpful to subtract half of the _x_ step from the _x_ coordinate. You can also use `Scroll.Relative.x` to scroll by an _x_ delta. Cheers!

**Comment by yamin8000** [comment]
> [Vico 2.0.0 Alpha 9](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.9) adds such functionality. To scroll to the position of a specific entry, use `Scroll.Absolute.x`, passing in the entry’s _x_ coordinate, which you can align via the `bias` parameter. Depending on the specifics of the desired result, it can be helpful to subtract half of the _x_ step from the _x_ coordinate. You can also use `Scroll.Relative.x` to scroll by an _x_ delta. Cheers!

I'm new to this library so pardon my rudimentary use of it, So I tried this but nothing happened, I don't know what's the significance of `autoScrollCondition` parameter though. After adding some data (~20 to ~23, probably based on my screen width) I need to scroll by gesture, and it doesn't auto-scroll.

```kotlin
val scrollState = rememberVicoScrollState(
        autoScrollCondition = AutoScrollCondition.OnModelSizeIncreased
    )

val yS = mutableListOf(0)
val xS = mutableListOf(0)

val modelProducer = CartesianChartModelProducer.build()
    LaunchedEffect(Unit) {
        repeat(Int.MAX_VALUE) {
            xS.add(xS.last() + 1)
            yS.add(Random.nextInt(0, 10))

            modelProducer.tryRunTransaction {
                lineSeries {
                    series(yS)
                }
            }
            scrollState.scroll(Scroll.Relative.x(xS.last().toFloat()))
            delay(500)
        }
    }


ProvideVicoTheme(theme = rememberM3VicoTheme()) {
                        CartesianChartHost(
                            scrollState = scrollState,
                            chart = rememberCartesianChart(
                                rememberLineCartesianLayer(),
                                startAxis = rememberStartAxis(),
                                topAxis = rememberTopAxis(),
                                endAxis = rememberEndAxis(),
                                bottomAxis = rememberBottomAxis(),
                            ),
                            modelProducer = modelProducer
                        )
                    }
```

**Reply by Gowsky** [reply]
Hello, @yamin8000. There's an issue in your code. You're using both automatic and programmatic scrolling in a manner that creates a conflict. Also, your approach with programmatic scrolling fails to consider the asynchronous nature of `Transaction`s. Here's a fixed version of the code:

```kt
val scrollState =
    rememberVicoScrollState(
        autoScroll = Scroll.Absolute.End,
        autoScrollCondition = AutoScrollCondition.OnModelSizeIncreased,
    )

val yS = mutableListOf(0)
val xS = mutableListOf(0)

val modelProducer = CartesianChartModelProducer.build()

LaunchedEffect(Unit) {
    repeat(Int.MAX_VALUE) {
        xS.add(xS.last() + 1)
        yS.add(Random.nextInt(0, 10))

        modelProducer.tryRunTransaction { lineSeries { series(yS) } }
        delay(500)
    }
}

CartesianChartHost(
    scrollState = scrollState,
    chart =
        rememberCartesianChart(
            rememberLineCartesianLayer(),
            startAxis = rememberStartAxis(),
            topAxis = rememberTopAxis(),
            endAxis = rememberEndAxis(),
            bottomAxis = rememberBottomAxis(),
        ),
    modelProducer = modelProducer
)
```

**Comment by Gowsky** [comment]
[Vico 2.0.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0) is now available.
